File: app/src/main/java/com/example/miodragmilosevic/nativetest/filters/javaFilters/OneColorFilter.java
package com.example.miodragmilosevic.nativetest.filters.javaFilters;

import android.graphics.Bitmap;
import android.graphics.Color;

import com.example.miodragmilosevic.nativetest.filters.ImageFilter;

import java.util.Random;

/**
 * Created by miodrag.milosevic on 12/15/2017.
 */

public class OneColorFilter extends ImageFilter {
    @Override
    protected int[] transformPixels(int[] pixelArray, int imageWidth, int imageHeight) {
        int red = 0;
        int green = 0;
        int blue = 0;
        Random random = new Random();
        int switchValue = random.nextInt(3);
        if (switchValue == 0) {
            red = 1;
        } else if (switchValue == 1) {
            green = 1;
        } else {
            blue = 1;
        }
        int A, R, G, B;
        for (int i = 0; i < pixelArray.length; i++) {
            A = Color.alpha(pixelArray[i]);
            R = Color.red(pixelArray[i]) * red;
            G = Color.green(pixelArray[i] * green);
            B = Color.blue(pixelArray[i]) * blue;
            pixelArray[i] = Color.argb(A, R, G, B);
        }
        return (pixelArray);
    }
}

File: app/src/main/cpp/OneColorFilter.cpp
//
// Created by miodrag.milosevic on 12/29/2017.
//

#include <stdlib.h>
#include "OneColorFilter.h"
#include "color.h"


void OneColorFilter::transformPixels(jint *pixelArray, jint imageWidth, jint imageHeight) {
    jint red = 0;
    jint green = 0;
    jint blue = 0;
    jint switchValue = rand() % 3;
    if (switchValue == 0) {
        red = 1;
    } else if (switchValue == 1) {
        green = 1;
    } else {
        blue = 1;
    }
    jint A, R, G, B;
    for (jint i = 0; i < imageHeight*imageWidth; i++) {
        A = Color::alpha(pixelArray[i]);
        R = Color::red(pixelArray[i]) * red;
        G = Color::green(pixelArray[i] * green);
        B = Color::blue(pixelArray[i]) * blue;
        pixelArray[i] = Color::argb(A, R, G, B);
    }
}
